pkg/scheduler/framework/plugins/podtopologyspread/plugin.go

```diff
@@ -141,9 +145,141 @@ func (pl *PodTopologySpread) EventsToRegister() []framework.ClusterEventWithHint
 		// an unschedulable Pod schedulable.
 		// - Delete. An unschedulable Pod may fail due to violating an existing Pod's topology spread constraints,
 		// deleting an existing Pod may make it schedulable.
-		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.All}},
-		// Node add|delete|updateLabel maybe lead an topology key changed,
+		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.All}, QueueingHintFn: pl.isSchedulableAfterPodChange},
+		// Node add|delete|update maybe lead an topology key changed,
 		// and make these pod in scheduling schedulable or unschedulable.
-		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.Delete | framework.UpdateNodeLabel}},
+		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.Delete | framework.Update}, QueueingHintFn: pl.isSchedulableAfterNodeChange},
 	}
 }
+
+func shouldNotBeAffectedByPodChange(incomingPod, targetPod *v1.Pod) bool {
+	return incomingPod.Spec.NodeName == "" || incomingPod.Namespace != targetPod.Namespace
+}
```

Comment:
I prefer this kind of functions to be a positive comparison instead of `NOT` comparison: 

```suggestion
// involvedInTopologySpreading checks if incomingPod is involved in the pod spread constraint of podWithSpreading.
func involvedInTopologySpreading(podWithSpreading, incomingPod *v1.Pod) bool {
	return incomingPod.Spec.NodeName != "" || incomingPod.Namespace == targetPod.Namespace
}
```

Reply:
`return incomingPod.Spec.NodeName != "" && incomingPod.Namespace == targetPod.Namespace`